rasa/nlu/featurizers/dense_featurizer/mitie_featurizer.py

```diff
@@ -17,41 +18,65 @@
 from rasa.utils.tensorflow.constants import MEAN_POOLING, POOLING
+from rasa.nlu.featurizers.dense_featurizer._mitie_featurizer import MitieFeaturizer
 
 if typing.TYPE_CHECKING:
     import mitie
 
+logger = logging.getLogger(__name__)
+
+# TODO: This is a workaround around until we have all components migrated to
+# `GraphComponent`.
+MitieFeaturizer = MitieFeaturizer
 
-class MitieFeaturizer(DenseFeaturizer):
-    @classmethod
-    def required_components(cls) -> List[Type[Component]]:
-        return [MitieNLP, Tokenizer]
 
-    defaults = {
-        # Specify what pooling operation should be used to calculate the vector of
-        # the complete utterance. Available options: 'mean' and 'max'
-        POOLING: MEAN_POOLING
-    }
+class MitieFeaturizerGraphComponent(DenseFeaturizer2, GraphComponent):
+    """A class that featurizes using Mitie."""
 
-    def __init__(self, component_config: Optional[Dict[Text, Any]] = None) -> None:
-        super().__init__(component_config)
+    @staticmethod
+    def get_default_config() -> Dict[Text, Any]:
+        """Returns the component's default config."""
+        return {
+            **DenseFeaturizer2.get_default_config(),
+            # Specify what pooling operation should be used to calculate the vector of
+            # the complete utterance. Available options: 'mean' and 'max'
+            POOLING: MEAN_POOLING,
+        }
 
-        self.pooling_operation = self.component_config["pooling"]
+    def __init__(
+        self,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> None:
+        """Instantiates a new `MitieFeaturizerGraphComponent` instance."""
+        super().__init__(config)
+        self.config = config
+        self.pooling_operation = self.config["pooling"]
 
     @classmethod
-    def required_packages(cls) -> List[Text]:
+    def create(
+        cls,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> "MitieFeaturizerGraphComponent":
+        """Creates a new untrained component (see parent class for full docstring)."""
+        return cls(config, model_storage, resource, execution_context)
+
+    @staticmethod
+    def required_packages() -> List[Text]:
+        """Any extra python dependencies required for this component to run."""
         return ["mitie", "numpy"]
 
     def ndim(self, feature_extractor: "mitie.total_word_feature_extractor") -> int:
+        """Returns the number of dimensions."""
         return feature_extractor.num_dimensions
 
-    def train(
-        self,
-        training_data: TrainingData,
-        config: Optional[RasaNLUModelConfig] = None,
-        **kwargs: Any,
-    ) -> None:
-
+    def train_process(self, training_data: TrainingData, **kwargs: Any,) -> None:
```

Comment:
I know the issue called it this, but we've been using `process_training_data` elsewhere.

Reply:
I think we need the `model: MitieModel` here like in `MitieIntentClassifierGraphComponent`. (and I don't think we need `kwargs` anymore.